galsim/correlatednoise.py

```diff
@@ -1491,17 +1525,21 @@ def __init__(self, variance, rng=None, scale=None, wcs=None, gsparams=None):
             if not isinstance(wcs, galsim.BaseWCS):
                 raise TypeError("wcs must be a BaseWCS instance")
             if not wcs.isUniform():
-                raise ValueError("Cannot provide non-local wcs")
+                raise ValueError("Cannot provide non-uniform wcs")
         elif scale is not None:
             wcs = galsim.PixelScale(scale)
         else:
             wcs = galsim.PixelScale(1.0)
 
-        # Need variance == xvalue(0,0)
-        # Pixel has flux of f/scale^2, so use f = variance * scale^2
-        image_pix = galsim.Pixel(scale=1.0, flux=variance * wcs.pixelArea(), gsparams=gsparams)
-        world_pix = wcs.toWorld(image_pix)
-        cf_object = galsim.AutoConvolve(world_pix)
-        _BaseCorrelatedNoise.__init__(self, rng, cf_object)
+        # Need variance == xvalue(0,0) after autoconvolution
+        # So the Pixel needs to have an amplitude of sigma at (0,0)
+        import math
```

Comment:
Out of curiosity I just did a timing test, and found that not only is ```math.sqrt``` faster (which I thought obviously had to the case - though on my system it's a factor of 1.8-2), but importing math AND doing ```math.sqrt``` is faster than ```bumpy.sqrt``` on its own.  That was pretty non-obvious to me.